solution/src/main/kotlin/com/cosmotech/solution/service/SolutionServiceImpl.kt

```diff
@@ -464,6 +537,26 @@ class SolutionServiceImpl(
     return SolutionAccessControl(rbacAccessControl.id, rbacAccessControl.role)
   }
 
+  override fun getSolutionFile(
+      organizationId: String,
+      solutionId: String,
+      fileName: String
+  ): Resource {
+    require(".." !in fileName) { "Invalid filename: '$fileName'. '..' is not allowed" }
+    val solution = getVerifiedSolution(organizationId, solutionId, PERMISSION_READ)
```

Comment:
Why use `getVerifiedSolution` here and `getSolution` in `listSolutionFiles`? Is there a difference?